sources/api/apiserver/src/server/mod.rs

```diff
@@ -342,9 +353,34 @@ async fn get_transaction(
     let transaction = transaction_name(&query);
     let datastore = data.ds.read().ok().context(error::DataStoreLockSnafu)?;
     let data = controller::get_transaction(&*datastore, transaction)?;
+
     Ok(SettingsResponse(data))
 }
 
+/// Get any pending settings in the given transaction, or the "default" transaction if unspecified.
+async fn get_transaction_v2(
+    query: web::Query<HashMap<String, String>>,
+    data: web::Data<SharedData>,
+) -> Result<SettingsResponseWithMetadata> {
+    let transaction = transaction_name(&query);
+    let datastore = data.ds.read().ok().context(error::DataStoreLockSnafu)?;
+    let settings = controller::get_transaction(&*datastore, transaction)?;
+    let transaction_metadata =
+        controller::get_transaction_metadata(&*datastore, transaction, None)?;
+
+    let mut metadata = HashMap::new();
+    for (key, value) in transaction_metadata {
+        let mut inner_map = HashMap::new();
+        for (inner_key, inner_value) in value {
+            inner_map.insert(inner_key.name().clone(), inner_value);
+        }
+        metadata.insert(key.name().clone(), inner_map);
+    }
```

Comment:
Carrying around deeply nested hashmaps and performing "on-the-spot" conversions like this can get messy fast.

Suggestion:
* Make a type like `SettingsMetadata` which implements `From<HashMap<Key, HashMap<Key, Value>>>`
* `SettingsMetadata` can `#[derive(Serialize, Deserialize)]` and use the `#[serde(transparent)]` to expose the inner nested hashmap.
* Modify `get_transaction_metadata` to return a `SettingsMetadata` struct.